core/services/relay/evm/relayer_extender.go

```diff
@@ -55,6 +57,8 @@ func NewLegacyChains(ctx context.Context, opts legacyevm.ChainRelayOpts) (result
 		}
 	}
 
+	// map with lazy initialization for the txm to access evm clients for different chain
+	var chainIDToClientMap = make(map[string]rollups.EVMClient)
 	for i := range enabled {
```

Comment:
When/how is `evmConfigs[i].IsEnabled()` set in the config? Is it possible to end up in a state where we need a client for `chainA` but we don't end up adding it to the map because `chainA.IsEnabled()` is `false`?

For example:
- we have a chain called  `chainA` (maybe a Layer 1)
- we have an L2 chain called `chainB` 
- `chainB` needs to call a `chainA` function to be able to calculate the DA gas price
- the EVM config for `chainB` has `enabled == true`, but for some reason the EVM config for `chainA` has `enabled == false`. If this is the case then we won't be able to use and call the `chainA` client in `chainB`'s DA gas estimator

Could this ever happen? Seems unlikely that we'd ever enable an L2 without supporting its corresponding L1, but I wanted to check just in case

Reply:
The `enabled` flag is provided by Toml config (true if not [provided](https://github.com/smartcontractkit/chainlink/blob/fcf7805ce477b13953e70ad76eb631feb647b65e/core/chains/evm/config/toml/config.go#L279C11-L279C17)), and not changed in runtime. So unless the config has `enabled=false` by mistake in the config, you should find all client from the map provided to the config. 

The user of the map `(NewL1GasOracle)` should return error when the target chain id is missing.